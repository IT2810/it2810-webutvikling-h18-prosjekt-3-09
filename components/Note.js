import React  from 'react';
import {
    StyleSheet,
    Text,
    View,
    TouchableOpacity
  } from "react-native";

export default class Note extends React.Component {
      render() { 
          const {id, title, data, deleteMethod} = this.props
          return ( 
              <View id={id} style={styles.note}>
                
                <Text style={styles.noteTitle}>{title}</Text>
                <Text style={styles.noteText}>{data}</Text>

                <TouchableOpacity 
                    onPress={deleteMethod} 
                    style={styles.noteDelete}
                    accessibilityLabel='Press to remove task from list'
                    >
                    <Text style={styles.noteDeleteText}>x</Text>
                </TouchableOpacity>

              </View>
           );
      }
  }

const styles = StyleSheet.create({
    note: {
    flex: 1,
    height:70,
    padding:5,
    margin:2,
    borderWidth:2,
    },
    noteTitle: {
    fontWeight: 'bold',
    },
    noteText: {
        paddingLeft: 20,
        borderLeftWidth: 10,
    },
    noteDeleteText: {
        color: 'black',
        fontWeight: 'bold',
    },
    noteDelete: {
    borderWidth:1,
    borderColor:'"rgba(0,0,0,0.2)"',
    alignItems:'center',
    justifyContent:'center',
    width:20,
    height:20,
    left:5,
    backgroundColor:'#e5547b',
    borderRadius:20,
    }
})