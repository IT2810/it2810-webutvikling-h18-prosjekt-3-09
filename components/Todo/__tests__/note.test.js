import React from "react";
import Note from "../Note";

import renderer from "react-test-renderer";
const tree = renderer.create(<Note title={"title"} />);

test("renders correctly", () => {
  expect(tree).toMatchSnapshot();
});

test("test me", () => {
  console.log(tree.toTree().instance.props);
});
