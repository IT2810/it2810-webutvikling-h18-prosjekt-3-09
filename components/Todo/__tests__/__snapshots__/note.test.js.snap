// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderColor": "#e1e8ee",
        "borderWidth": 1,
        "margin": 15,
        "marginBottom": 0,
        "padding": 15,
        "shadowColor": "rgba(0,0,0, .2)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      },
      undefined,
      Object {
        "borderRadius": 6,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "flexGrow": 1,
            "fontWeight": "bold",
          }
        }
      >
        title
      </Text>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel="Press to remove task from list"
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            false,
            false,
            false,
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#e5547b",
                "fontSize": 24,
              },
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#e1e8ee",
            "height": 1,
          },
          Object {
            "backgroundColor": "#ddd",
            "marginBottom": 5,
            "marginTop": 5,
          },
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "marginBottom": 10,
        }
      }
    />
    <View
      onPress={undefined}
      style={
        Array [
          Object {
            "margin": 7,
          },
          Object {
            "borderRadius": 16,
            "height": 28,
            "width": 28,
          },
          Object {
            "shadowColor": "rgba(0,0,0, .4)",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      underlayColor="white"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#f50",
              "fontSize": 12,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
